crates/wasmtime/src/module/serialization.rs

```diff
@@ -329,7 +329,7 @@ impl<'a> SerializedModule<'a> {
         Ok(bytes)
     }
 
-    pub fn from_bytes(bytes: &[u8]) -> Result<Self> {
+    pub fn from_bytes(bytes: &[u8], check_version: bool) -> Result<Self> {
```

Comment:
I think the `check_version = false` version should be unsafe as it is much easier to get wasmtime to interpret caches from a different wasmtime version to be interpreted incorrectly in such a way that you get UB.

Reply:
I disagree, the `deserialize` function is already `unsafe`, so this should just add caveat there to the documentation.